addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/launch-product/launch-product.js

```diff
@@ -356,6 +359,34 @@ class LaunchProduct extends StepBase {
     }
     return result.LaunchPathSummaries[0];
   }
+
+  /**
+   * Method to check if the resolved input parameter contained a static namespace param. If so, this method augments the
+   * namespace to begin with 'analysis-' for permissions purposes (if it does not already start with that) and to end with 
+   * a unique datetime string so Cloudformation doesn't make create duplicate stacks on seperate deployments of the same
+   * workspace configuration. 
+   * 
+   * @param resolvedInputParams 
+   * @param datetime 
+   * @returns {Promise<{Value: string, Key: string}[]>}
+   */
+  async checkNamespace(resolvedInputParams, datetime){
+    const namespaceIndex = resolvedInputParams.findIndex(element => element.Key === 'Namespace');
+    let namespaceParam = resolvedInputParams[namespaceIndex].Value;
+
+    // Check to make sure the resolved namespace variable begins with 'analysis-' so our templates will allow it
+    if(!namespaceParam.startsWith('analysis-')){
+      namespaceParam = 'analysis-' + namespaceParam;
+    }
+
+    // Check to make sure the resolved namespace variable ends with a unique datetime string so it will be unique for each deployment of a configuration with a static namespace
+    if(namespaceParam.split('-').pop() !== datetime.toString()){
```

Comment:
Correct me if I'm wrong, but I think this statement will always evaluate to `true`. I don't think a user can pass in a `dateTime` value with the exact time that this code segment is executed. We probably always want to use the `dateTime` value being passed in as the suffix.

Reply:
It will not evaluate to true if the resolveVarExpressions() function call does not change resolvedVars.namespace I believe. I had tested it and with a dynamic placeholder the code never entered into this if branch.